pandas/tests/reshape/test_tile.py

```diff
@@ -512,6 +512,17 @@ def f():
         tm.assert_numpy_array_equal(
             mask, np.array([False, True, True, True, True]))
 
+    def test_cut_read_only(self):
+        # issue 18773
+        readonly = np.arange(0, 100, 10)
+        readonly.flags.writeable = False
+
+        mutable = np.arange(0, 100, 10)
+
+        hundred_elements = np.arange(100)
+        tm.assert_categorical_equal(cut(hundred_elements, readonly),
```

Comment:
you can paramaterize on writeable here.

Reply:
I think this is what you're asking. I added a parameterized test that tests the three combinations (Readonly, Mutable) (Mutable, Mutable) (Readonly, Readonly). Let me know if that's not what you were thinking. Thanks @jreback